galite-core/src/main/kotlin/org/kopi/galite/report/VReportCommand.kt

```diff
@@ -15,6 +15,99 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
+
 package org.kopi.galite.report
 
-class VReportCommand 
\ No newline at end of file
+import org.kopi.galite.print.DefaultPrintManager
+import org.kopi.galite.print.PrintManager
+import org.kopi.galite.print.Printable
+import org.kopi.galite.visual.VActor
+import org.kopi.galite.visual.VCommand
+import org.kopi.galite.visual.ActionHandler
+import org.kopi.galite.visual.Action
+import org.kopi.galite.visual.PrinterManager
+
+class VReportCommand(val report: VReport, actor: VActor) : VCommand(0xFFFF, null, actor, actor.number, actor.actorIdent), ActionHandler {
+  /**
+   * Returns the actor
+   */
+  override fun setEnabled(enabled: Boolean) {
+    if (actor != null) {
+      actor!!.isEnabled = enabled
+      actor!!.number = trigger
+      actor!!.handler = this
+    }
+  }
+
+  /**
+   * Performs the appropriate action asynchronously.
+   * You can use this method to perform any operation out of the UI event process
+   *
+   * @param    action        the action to perform.
+   * @param    block        This action should block the UI thread ?
+   */
+  override fun performAction(action: Action, block: Boolean) {
+    report.performAction(action, block)
+    /*try {
+      executeVoidTrigger(getTrigger());
+    } catch (Exception e) {
+      e.printStackTrace();
+    }*/
+  }
+
+  /**
+   * Performs the appropriate action asynchronously.
+   * You can use this method to perform any operation out of the UI event process
+   *
+   * @param    action        the action to perform.
+   */
+  override fun performAsyncAction(action: Action) {
+    report.performAsyncAction(action)
+    /*try {
+      executeVoidTrigger(getTrigger());
+    } catch (Exception e) {
+      e.printStackTrace();
+    }*/
+  }
+
+  /**
+   * Performs a void trigger
+   *
+   * @param    type    the number of the trigger
+   */
+  override fun executeVoidTrigger(type: Int) {
+    when (type) {
+      Constants.CMD_QUIT -> report.close()
+      Constants.CMD_PRINT -> {
+        val pm: PrintManager = DefaultPrintManager.getPrintManager()
+        pm.print(report,
+                report as Printable,
```

Comment:
remove `as Printable` since `VReport` is a `Printable`
then you can remove unused import